File: backend/src/main/java/com/mulesoft/arc/arcdatastore/backend/models/ErrorResponse.java
package com.mulesoft.arc.arcdatastore.backend.models;

/**
 * Error report from the server.
 */
public class ErrorResponse {
    /**
     * Status code.
     */
    public int code;
    /**
     * Message explaining error.
     */
    public String message;

    public ErrorResponse() {}
}

File: models/error-response.js
'use strict';

/**
 * Error report from the server.
 */
class ErrorResponse {
  /**
   * To construct error response provide a status `code` and the reason `message`.
   *
   * @param {Number} code Status code associated with this response.
   * @param {String} message A reason message.
   */
  constructor(code, message) {
    this.code = code || 400;
    this.message = message || 'Unknown error ocurred';
  }
}

module.exports.ErrorResponse = ErrorResponse;
